accord-core/src/main/java/accord/utils/async/AsyncChains.java

```diff
@@ -423,6 +424,16 @@ public static <V> AsyncChain<V> reduce(AsyncChain<V> a, AsyncChain<V> b, BiFunct
         return new Reduce<>(Lists.newArrayList(a, b), reducer);
     }
 
+    public static <A, B> AsyncChain<B> reduce(List<? extends AsyncChain<? extends A>> chains, B zero, BiFunction<B, ? super A, B> reducer)
+    {
+        switch (chains.size())
+        {
+            case 0: return AsyncChains.success(zero);
+            case 1: return chains.get(0).map(a -> reducer.apply(zero, a));
+        }
+        return new ReduceWithZero<>(chains, zero, reducer);
```

Comment:
I don't think this is needed. The empty case is already handled above, and the only usage of this doesn't need a special initial value

Reply:
fair, we know it has 2, I think I did this trying to find a way to merge `Reduce` to handle both cases but after awhile I migrated to use its own class; ill simplify

Reply:
trying to resolve this comment and I remember why it is needed; `zero` is type `B` but the chains are of type `A`, so can not reduce without keeping the `zero`